### ISSUE TITLE
Mails not working ?

### ISSUE THREAD

**Issue by Tusharsb** [johanek/johanek-redmine#62]
I deployed the puppet on a digitalocean instance and my redmine notification emails do not seem to be working. I spent few hrs toubleshooting this. But couldnt get any hint why.. anyone else facing problem of emails not getting sent ?

**Comment by UnrealQuester** [comment]
If I remember correctly, there should be a email notification page somewhere in the admin menu with a button to send a test mail. Can you try that and see if it gives you an error?
Also, does your server need any kind of authentication?
You might also want to take a look at http://www.redmine.org/projects/redmine/wiki/EmailConfiguration

**Comment by Tusharsb** [comment]
Couldnt find an 'email test' button in settings page. Find its screenshot below
![screenshot-projects smarter codes 2015-10-12 11-34-48](https://cloud.githubusercontent.com/assets/1430472/10422263/8b217916-70d5-11e5-9740-c39694945463.png)

I had tried to follow the steps at http://www.redmine.org/projects/redmine/wiki/EmailConfiguration before. After your recommendation, I went through them once again. Made a small improvement in my configuration.yml by mimicing the examples mentioned in there. Now my configuration.yml reads like this 
```
default:
        email_delivery:
          delivery_method: :smtp
          smtp_settings:
              address: localhost
              port: 25
              domain: [masked host name]
```
basically I added the 'email_delivery' header under 'default' header. I also tried changing 'default' header to 'production'. But none worked.

Note that - if I do telnet to 25 port (postfix running on my machine) and manually handshake and try sending email - I am successfully able to receive emails from postfix running there. So the email server must be running

If I can find some 'test mail' button like you said, and see some error logs. I did checked the log file at ```/path/to/redmine/log/production.log``` but it is 0 bytes long (empty). So couldnt get any clue from there as well

**Comment by UnrealQuester** [comment]
The button is in the lower right of your screenshot. It's called "Send a test email".
Is the production log writeable by your apache process?

**Comment by Tusharsb** [comment]
Thanks! I should have seen more carefully :-)

Upon pressing the 'Send a test email' link, it did showed me a beautiful error (saying '')

While I eventually resolved by following the steps at http://www.redmine.org/boards/2/topics/30851

Thanks a lot for your help UnrealQuester. To fix this I changed my configuration.yml to the following 
```
  GNU nano 2.2.6           File: configuration.yml                              

default:
  email_delivery:
    delivery_method: :smtp
    smtp_settings:
      address: localhost
      port: 25
      domain: [masked hostname]
      openssl_verify_mode: 'none'
      authentication: :none
```
observe the last 2 new lines. I think we should be added these 2 lines in puppet if others are also facing the same problem ?

**Comment by UnrealQuester** [comment]
I do not think we should make something that adds a potential security risk the default.

**Comment by jqueuniet** [comment]
I wanted to chime in since I ran into the same problem. Redmine was configured to talk with localhost on port 25, and obviously localhost isn't considered valid, neither with the built-in selfsigned certificate or a real one. The specific error message was looking like this.

 ````
Email delivery error: hostname "localhost" does not match the server certificate
 ````

I solved my problem with this configuration.

```
default:
  delivery_method: :smtp
  smtp_settings:
      address: localhost
      port: 25
      domain: *********.***
      enable_starttls_auto: false
 ````
Adding a special case for localhost would be appreciated.

**Comment by noxmetus** [comment]
> ```
>       enable_starttls_auto: false
> ```
> 
This will send your emails not encrypted.
